test/powershell/Modules/Microsoft.PowerShell.Core/Job.Tests.ps1

```diff
@@ -40,10 +56,94 @@ Describe "Job Cmdlet Tests" -Tag "CI" {
+        It "Stop-Job will stop a job" {
             Stop-Job -id $j.id
-            $j.Status |Should be Stopped
+            $j.State | should be Stopped
+        }
+        It "Remove-Job will not remove a running job" {
+            $id = $j.Id
+            remove-job $j -ErrorAction SilentlyContinue
+            $job = Get-Job -Id $id
+            $job | should be $j
+        }
+        It "Remove-Job -Force will remove a running job" {
+            $id = $j.Id
+            Remove-Job $j -Force
+            $job = Get-Job -Id $id -ErrorAction SilentlyContinue
+            $job | should be $null
+        }
+    }
+    Context "Retrieving partial output from jobs" {
+        BeforeAll {
+            function getResults($job, $n, $keep)
+            {
+                $results = @()
+
+                # $count lets us bail out after 5 minutes, avoiding an endless loop
+                for ($count = 0; $results.Count -lt $n; $count++)
+                {
+                    if ($count -eq 10000)
+                    {
+                        # It's been 5 minutes and we still have collected enough results!
+                        throw "Receive-Job behaves suspiciously: Cannot receive $n results in 5 minutes."
+                    }
+
+                    # sleep for 300 ms to allow data to be produced
+                    Start-Sleep -Milliseconds 300
+
+                    if ($keep)
+                    {
+                        $results = Receive-Job -Keep $job
+                    }
+                    else
+                    {
+                        $results += Receive-Job $job
+                    }
+                }
+
+                #Write-Verbose -Verbose "Received $($results.Count) results in $($count * 300) milliseconds"
+
+                return $results
+            }
+
+            # call this only within an It block
+            function checkContent($array)
+            {
+                for ($i=1; $i -lt $array.Count; ++$i)
+                {
+                    $array[$i] | should be ($array[$i-1] + 1)
+                }
+            }
+
+        }
+        BeforeEach {
+            $j = start-job -scriptblock { $count = 1; while ($true) { Write-Output ($count++); Start-Sleep -Milliseconds 30 } }
+        }
+        AfterEach {
+            Get-Job | Stop-Job
+            Get-Job | Remove-Job -force
+        }
+
+        It "Receive-Job will retrieve partial output" {
+            $result1 = getResults $j 5 $false
+            $result2 = getResults $j 5 $false
+            checkContent ($result1 + $result2)
+        }
+        It "Receive-Job will retrieve partial output, including -Keep results" {
+            $result1 = getResults $j 5 $true
+            $result2 = getResults $j ($result1.Count + 5) $false
+            checkContent $result1
+            for ($i=0; $i -lt $result1.count; ++$i)
+            {
+                $result1[$i] | should be $result2[$i]
+            }
+            checkContent $result2
```

Comment:
Both `checkContent` should be removed because this has already been checked in the previous test.

And it makes no sense to make 'Should Be' for every array member. We could use `Compare-Object` here.